src/components/ScreenCharacterSelect/index.tsx

```diff
@@ -18,10 +18,11 @@ import priestIcon from "./priest.png";
 const CLASSES: HeroClass[] = ["paladin", "warlock", "rogue", "mage", "hunter", "druid", "warrior", "shaman", "priest"];
 
 interface Props {
-  onConfirmSelection: () => void;
+  onConfirmSelection: (faction: Faction, heroClass: HeroClass) => void;
 }
 
 const CharacterSelectScreen = (props: Props) => {
+  // TODO: Use context directly here?
```

Comment:
Yeah that's what I decided too, just forgot to delete this. Actually, I think this would make a lot more sense if we didn't initialize the provider until after we finished character select